async.js

```diff
@@ -0,0 +1,33 @@
+"use strict"
+
+/* globals Promise:true */
+
+function makeAsyncApi(api) {
+    return Object.keys(api).reduce(function (asyncApi, method) {
+        if (isWrapableMethod(method, api)) {
+            asyncApi[method] = createAsyncMethodProxy(api[method])
+        }
+
+        return asyncApi
+    }, {})
+}
+
+function isWrapableMethod(method, api) {
+    return method !== "async" &&
+        method !== "AssertionError" &&
+        typeof api[method] === "function"
+}
+
+function createAsyncMethodProxy(method) {
+    return function proxy(expr) {
+        var rest = [].slice.call(arguments, 1)
+
+        return Promise.resolve(expr)
+            .then(function (value) {
+                method.apply(undefined, [value].concat(rest))
+            }
+        )
+    }
+}
+
+module.exports = makeAsyncApi
```

Comment:
Could you instead just call `makeAsyncApi` repeatedly for each other primary `clean-assert` export, so people can just use `require("clean-assert/async")`?

You can basically copy/paste most of the `index.js` exports and do a giant single-pass multi-select (if your editor allows it) to wrap them all.

Sorry if I hadn't made that clear initially, but that was my eventual intent.

Oh, and you'll have to change several imports accordingly.